sentry/src/routes.rs

```diff
@@ -72,53 +108,113 @@
 //!
-//! Request Body: [`primitives::sentry::campaign_create::CreateCampaign`] (json)
 //!
-//!  `POST /v5/campaign/:id/close` (auth required)
+//! Request body (json): [`CreateCampaign`][primitives::sentry::campaign_create::CreateCampaign]
 //!
-//! todo
+//! Response: [`Campaign`]
+//!
+//! - [`POST /v5/campaign/:id`](campaign::update_campaign::handle_route) (auth required)
+//!
+//! Modify the [`Campaign`]. Request must be sent by the [`Campaign.creator`](primitives::Campaign::creator).
+//!
+//! **Authentication is required** to validate [`Campaign.creator`](primitives::Campaign::creator) == [`Auth.uid`](crate::Auth::uid)
+//!
+//! Request body (json): [`ModifyCampaign`][primitives::sentry::campaign_modify::ModifyCampaign]
+//!
+//! Response: [`Campaign`]
+//!
+//! - [`POST /v5/campaign/:id/close`](campaign::close_campaign) (auth required)
+//!
+//! Close the campaign.
+//!
+//! Request must be sent by the [`Campaign.creator`](primitives::Campaign::creator).
+//!
+//! **Authentication is required** to validate [`Campaign.creator`](primitives::Campaign::creator) == [`Auth.uid`](crate::Auth::uid)
+//!
+//! Closes the campaign by setting [`Campaign.budget`](primitives::Campaign::budget) so that `remaining budget = 0`.
+//!
+//! Response: [`SuccessResponse`]
 //!
 //! ## Analytics
 //!
-//! - `GET /v5/analytics`
+//! - [`GET /v5/analytics`](get_analytics)
+//!
+//! Allowed keys: [`AllowedKey::Country`][primitives::analytics::query::AllowedKey::Country] & [`AllowedKey::AdSlotType`][primitives::analytics::query::AllowedKey::AdSlotType]
+//!
+//! - [`GET /v5/analytics/for-publisher`](get_analytics) (auth required)
 //!
-//! todo
+//! Returns all analytics where the currently authenticated address [`Auth.uid`](crate::Auth::uid) is a **publisher**.
 //!
-//! - `GET /v5/analytics/for-publisher` (auth required)
+//! All [`ALLOWED_KEYS`] are allowed for this route.
 //!
-//! todo
 //!
-//! - `GET /v5/analytics/for-advertiser` (auth required)
+//! - [`GET /v5/analytics/for-advertiser`](get_analytics) (auth required)
 //!
-//! todo
+//! Returns all analytics where the currently authenticated address [`Auth.uid`](crate::Auth::uid) is an **advertiser**.
 //!
-//! - `GET /v5/analytics/for-admin` (auth required)
+//! All [`ALLOWED_KEYS`] are allowed for this route.
 //!
-//! todo
+//! - [`GET /v5/analytics/for-admin`](get_analytics) (auth required)
 //!
+//! Admin access to the analytics with no restrictions on the keys for filtering.
+//!
+//! All [`ALLOWED_KEYS`] are allowed for admins.
+//!
+//! Admin addresses are configured in the [`Config.admins`](primitives::Config::admins)
+//!
+//! [`Adapter`]: adapter::Adapter
+//! [`Address`]: primitives::Address
+//! [`AllowedKey`]: primitives::analytics::query::AllowedKey
+//! [`ALLOWED_KEYS`]: primitives::analytics::query::ALLOWED_KEYS
+//! [`ApproveState`]: primitives::validator::ApproveState
+//! [`Accounting`]: crate::db::accounting::Accounting
+//! [`AccountingResponse`]: sentry::db::accounting::Accounting
```

Comment:
Should be `primitives::src::sentry::AccountingResponse`

Reply:
fixed.